Makefile

```diff
@@ -83,5 +83,35 @@ shellcheck:
 shfmt:
 	shfmt -w -i 4 test scripts
 
+# Python test targets
+.PHONY: test-python-setup test-python test-concurrency test-integration test-all
+
+# Setup Python test environment
+test-python-setup:
+	@echo "Setting up Python test environment..."
+	pip3 install --break-system-packages -r tests/requirements.txt
```

Comment:
`--break-system-packages` seems bad? We should be doing this in a venv, or at least have an explanation for why this is here